tests/directory/BUILD

```diff
@@ -0,0 +1,104 @@
+load("@rules_testing//lib:analysis_test.bzl", "analysis_test")
+load(
+    ":directory_test.bzl",
+    "directory_with_self_srcs_test",
+    "generated_root_test",
+    "no_srcs_root_test",
+    "outside_testdata_test",
+    "source_and_generated_root_test",
+    "source_root_test",
+)
+load(
+    ":subdirectory_test.bzl",
+    "nonexistent_subdirectory_test",
+    "subdirectory_as_file_test",
+    "subdirectory_of_file_test",
+    "subdirectory_test",
+)
+
+analysis_test(
```

Comment:
Having analysis_test in BUILD file, targets in test_data and assertions in directory_test.bzl file makes the tests hard to follow. Arrange, act and assert parts are in different locations

The idea behind analysis_test was that you can create all of the in the same file. That way the tests should be easier to follow. See: https://rules-testing.readthedocs.io/en/latest/analysis_tests.html